frontend/src/components/nodes/Nodes.tsx

```diff
@@ -1,61 +1,101 @@
+import { validators as VA } from "near-api-js";
 import React from "react";
 
 import NodesApi, * as N from "../../libraries/explorer-wamp/nodes";
 
-import autoRefreshHandler from "../utils/autoRefreshHandler";
-import FlipMove from "../utils/FlipMove";
-
 import NodeRow from "./NodeRow";
+import ValidatorRow from "./ValidatorRow";
+import ProposalRow from "./ProposalRow";
+import PaginationSpinner from "../utils/PaginationSpinner";
 
-import { OuterProps } from "../accounts/Accounts";
-
-interface Props extends OuterProps {
+interface Props {
   role: string;
 }
-export default class extends React.Component<Props> {
-  static defaultProps = {
-    count: 15,
-  };
 
-  fetchNodes = async (count: number, paginationIndexer?: string) => {
-    return await new NodesApi().getNodes(
-      count,
-      this.props.role,
-      paginationIndexer
-    );
+interface State {
+  onlineNodes?: N.NodeInfo[];
+  validators?: N.Validating[];
+  proposals?: N.Proposal[];
+}
+export default class extends React.Component<Props, State> {
+  state: State = {};
+
+  getNonValidatingNodes = async () => {
+    return await new NodesApi().getOnlineNodes();
   };
 
-  componentDidUpdate(prevProps: any) {
-    if (this.props.role !== prevProps.role) {
-      this.autoRefreshNodes = autoRefreshHandler(Nodes, this.config);
+  signNewValidators = (newValidators: any) => {
+    for (let i = 0; i < newValidators.length; i++) {
+      newValidators[i].new = true;
     }
-  }
+    return newValidators;
+  };
 
-  config = {
-    fetchDataFn: this.fetchNodes,
-    count: this.props.count,
-    category: "Node",
+  signRemovedValidators = (removedValidators: any) => {
+    for (let i = 0; i < removedValidators.length; i++) {
+      removedValidators[i].removed = true;
+    }
+    return removedValidators;
   };
 
-  autoRefreshNodes = autoRefreshHandler(Nodes, this.config);
+  getNodes = async () => {
+    if (this.props.role === "validators") {
+      let nodes = await new NodesApi().queryNodeRpc();
+      let currentValidators = nodes.current_validators;
+      let nextValidators = nodes.next_validators;
+      let diffEpochValidators = VA.diffEpochValidators(
+        currentValidators,
+        nextValidators
+      );
+      console.log(diffEpochValidators);
+      let newValidators = this.signNewValidators(
+        diffEpochValidators.newValidators
+      );
+      this.signRemovedValidators(diffEpochValidators.removedValidators);
```

Comment:
From the call-sight, it seems that these two methods operate differently while they are almost identical. If you need to change the objects in place, I suggest you do that instead of making an illusion that a new array is returned from `signNewValidators`.

```suggestion
      const { newValidators, removedValidators } = nearApi.validators.diffEpochValidators(
        currentValidators,
        nextValidators
      );
      this.signNewValidators(newValidators);
      this.signRemovedValidators(removedValidators);
```